components/eamxx/src/physics/ace/eamxx_ace_process_interface.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef EAMXX_ACE_PROCESS_INTERFACE_HPP
+#define EAMXX_ACE_PROCESS_INTERFACE_HPP
+
+#include <torch/script.h>
+
+#include <Kokkos_Core.hpp>
+
+#include "share/atm_process/atmosphere_process.hpp"
+#include "share/grid/remap/abstract_remapper.hpp"
+
+namespace scream {
+
+class ACE : public AtmosphereProcess {
+ public:
+  ACE(const ekat::Comm &comm, const ekat::ParameterList &params);
+
+  AtmosphereProcessType type() const override {
+    return AtmosphereProcessType::Physics;
+  }
+
+  std::string name() const override { return "ace"; }
+
+  void set_grids(
+      const std::shared_ptr<const GridsManager> grids_manager) override;
+
+#ifndef KOKKOS_ENABLE_CUDA
+ protected:
+#endif
+  void initialize_impl(const RunType run_type) override;
+  void run_impl(const double dt) override;
+  void finalize_impl() override;
+
+  // need three grids (lat-lon, point-grid, surface-coupling)
+  std::shared_ptr<const AbstractGrid> m_ll_grid, m_pt_grid, m_sc_grid;
+
+  // need tensors for input and output
+  // TODO: share pointers instead
+  torch::Tensor m_input_tensor, m_output_tensor;
```

Comment:
E.g., you could do

```c++
// in the hpp file
struct TorchData;

class ACE : public AtmosphereProcess {
...

  std::shared_ptr<TorchData> m_torch_data;
  ...
};

// In the cpp file
#include <torch/script.h>

struct TorchData {
   torch::Tensor input_tensor;
   torch::Tensor output_tensor;
   
   c10::InferenceMode guard;
   torch::jit::script::Module  module;
}
...

ACE::ACE (...) 
{
  m_torch_data = std::make_shared<TorchData>();
}
```
Later, simply prepend `m_torch_data->` to all the torch thingies you use.